android/app/src/main/java/org/fossasia/openevent/core/about/AboutFragmentViewModel.java

```diff
@@ -58,14 +60,15 @@ public AboutFragmentViewModel() {
     }
 
     public LiveData<List<Object>> getBookmarkedSessions() {
-        if (sessions == null) {
-            LiveRealmData<Session> sessionLiveRealmData = RealmDataRepository.asLiveData(realmRepo.getBookMarkedSessions());
-            sessions = Transformations.map(sessionLiveRealmData, this::getSessionsList);
-        }
-        return sessions;
+        if (filterableRealmLiveData == null)
+            filterableRealmLiveData = RealmDataRepository.asFilterableLiveData(realmRepo.getSessions());
+        Predicate<Session> predicate = Session::getIsBookmarked;
+        filterableRealmLiveData.filter(predicate);
+
+        return Transformations.map(filterableRealmLiveData, this::getSessionsList);
```

Comment:
Pulling in all sessions and then filtering is very inefficuent

Reply:
@iamareebjamal this is inefficient but I think this should be expected behaviour. 
If some session is bookmarked (which was not before) than changelistener of realm object will not be triggered because it was not in source set.

When we will change to room than I think
it will be automatically taken care by room.

I've changed to this because this is ideal behaviour(changelistener call each time) and it will be easier to change to room. we will not need to change any other logic other than getBookmarkedSessions.

Reply:
Sorry, my mistake. I've changed it to getBookmarkedSession and it worked because of change listener also notify about data changes in all sessions not only on received sessions.